src/k8s/pkg/snap/util/node.go

```diff
@@ -0,0 +1,38 @@
+package snaputil
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/canonical/k8s/pkg/utils"
+)
+
+func IsWorker(snap snap.Snap) (bool, error) {
+	return utils.FileExists(path.Join(snap.LockFilesDir(), "worker"))
+}
+
+func MarkAsWorkerNode(snap snap.Snap) error {
+	fname := path.Join(snap.LockFilesDir(), "worker")
+	lock, err := os.Create(fname)
+	defer lock.Close()
+	if err != nil {
+		return fmt.Errorf("failed to mark node as worker: %w", err)
+	}
+	if err := os.Chown(fname, snap.UID(), snap.GID()); err != nil {
+		return fmt.Errorf("failed to chown %s: %w", fname, err)
+	}
+	if err := os.Chmod(fname, 0700); err != nil {
+		return fmt.Errorf("failed to chmod %s: %w", fname, err)
+	}
+	return nil
+}
+
+func UnmarkAsWorkerNode(snap snap.Snap) error {
```

Comment:
just a nit, "unmark as worker node" does not read very well to me. could we instead have a single `func MarkWorkerNode(snap snap.Snap, mark bool) error`?